### PR TITLE
Rewrite Alert karma test for jest

### PR DESCRIPTION
**Pull request by meckila-navapbc**
Connects #14754

### Description
Rewrite Karma Alert Test for Jest

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Jest test for Alert
- [ ] Remove Alert Karma test

### CODE REVIEW

**Review by jcq — CHANGES_REQUESTED** [review]
Looking really good! I'd say we should add a snapshot to each test case just to be comprehensive, as well as a call to `jest-axe` for each as well

File: client/test/app/components/Alert.test.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    };
+
+    const { container } = render(<Alert {...props} />);
+
+    const results = await axe(container);
+
+    expect(results).toHaveNoViolations();
+  });
+
+  describe('shows correct alert type', () => {
```
Comment (jcq) [inline client/test/app/components/Alert.test.js:33]: Actually, this is a prime candidate for using Jest's `describe.each` functionality ([see docs](https://jestjs.io/docs/en/api#describeeachtablename-fn-timeout)).

You can probably do something like this:

```js
describe('Alert', () => {
  const title = 'This is a title';
  const types = ['error', 'info', 'success', 'warning'];

  describe.each(types)('alert type: %s', (type) => {
    const props = { title, type };

    it(`renders correctly for ${type}`, () => {
      const component = render(<Alert {...props} />);

      expect(component).toMatchSnapshot();
    });

    it(`passes a11y testing for ${type}`, async () => {
      const { container } = render(<Alert {...props} />);

      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
```

You could actually likely restructure things to put all of your tests in there, and really DRY out your code (which would probably make CodeClimate happy)

File: client/test/app/components/Alert.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import Alert from 'app/components/Alert';
+import { axe } from 'jest-axe';
+
+describe('Alert', () => {
+  const title = 'This is a title';
+  const message = 'This';
+  const types = ['error', 'info', 'success', 'warning'];
+
+  describe.each(types)(' type: %s', (type) => {
+    const props = { title, message, type };
```
Comment (jcq) [inline client/test/app/components/Alert.test.js:12]: The DRYing of the code is great! 

The only issue is that you ended up losing having a test that covers a version w/o the `message` prop

**Review by jcq — APPROVED** [review]
Looking really good! I think you want to add back one set of tests for the variation where `message` prop is undefined, but other than that should be GTG!

File: client/test/app/components/Alert.test.js (hunk 2 above)
Comment (meckila-navapbc) [reply client/test/app/components/Alert.test.js:19]: Noted. Added test to check when message is undefined

**Pull request merged**